agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -1599,6 +1603,21 @@ bool main_thread::handle_1905_discovery_query(ieee1905_1::CmduMessageRx &cmdu_rx
     return true;
 }
 
+bool main_thread::handle_1905_higher_layer_data_message(ieee1905_1::CmduMessageRx &cmdu_rx)
+{
+    const auto mid = cmdu_rx.getMessageId();
+    LOG(DEBUG) << "Received HIGHER_LAYER_DATA_MESSAGE , mid=" << std::hex << int(mid);
+
+    // build ACK message CMDU
+    auto cmdu_tx_header = cmdu_tx.create(mid, ieee1905_1::eMessageType::ACK_MESSAGE);
+    if (!cmdu_tx_header) {
+        LOG(ERROR) << "cmdu creation of type ACK_MESSAGE, has failed";
+        return false;
+    }
+    LOG(DEBUG) << "sending ACK message back to controller";
```

Comment:
Please also add the MID value to the debug log

Reply:
Ok

Reply:
When we set MID=0, the `transport service` allocates a new MID value which is incremented by one from the last one is used.
I think that for each ACK message we should add the message type to which we are "ack'ing" and the MID value, so it will be easy to see in the log of the agent that ACK was sent, and in the controller that ACK for the same message was received.

This should have been a (nitpick) anyway, we can refactor that later on.